lib/auth/grpcserver.go

```diff
@@ -51,11 +52,27 @@ import (
 	_ "google.golang.org/grpc/encoding/gzip" // gzip compressor for gRPC.
 )
 
-var heartbeatConnectionsReceived = prometheus.NewCounter(
-	prometheus.CounterOpts{
-		Name: teleport.MetricHeartbeatConnectionsReceived,
-		Help: "Number of times auth received a heartbeat connection",
-	},
+var (
+	heartbeatConnectionsReceived = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: teleport.MetricHeartbeatConnectionsReceived,
+			Help: "Number of times auth received a heartbeat connection",
+		},
+	)
+	watcherEventsEmitted = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: teleport.MetricWatcherEventsEmitted,
+			Help: "Number of watcher events emitted",
+		},
+		[]string{"resource", "size"},
+	)
```

Comment:
If I understand correctly, this creates a separate counter for each unique message size, correct?  This does not seem scaleable to me.  E.g. a cluster with many thousands of nodes, each with a slightly different label set, might result in what would effectively be a counter per node, right?  I know the backend limits the maximum resource size, so the vector technically can't grow _indefinitely_, but it feels pretty close.

 Maybe a histogram or some other "bucketing" strategy could be used?

Reply:
Switched to using a HistogramVec